src/mscorlib/shared/System/Buffers/ArrayPool.cs

```diff
@@ -2,8 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Runtime.CompilerServices;
+
 namespace System.Buffers
 {
+    internal static class ArrayPools
+    {
+        // Shared specialized ArrayPool<T> types created in non-generic class 
+        // to assist in devirtualization of ArrayPool<T>.Shared.
+        // The type in generic static field expansion is complex and 
+        // not picked up at AoT/crossgen time so won't currently devirtualize 
+        // when direct fields of ArrayPool<T> as part of crossgen.
+        // The general T pool should however devirtualize at Jit time.
+        public static TlsOverPerCoreLockedStacksArrayPool<byte> BytePool { [MethodImpl(MethodImplOptions.AggressiveInlining)] get; } =
```

Comment:
This should not be needed after https://github.com/dotnet/coreclr/pull/17078

Reply:
Only changes needed would be the `.Length` => `length` changes? (due to https://github.com/dotnet/coreclr/issues/15783)

Reply:
Its stored in a `ArrayPool` type so the Jit can't workout its type; it needs to be stored in a `TlsOverPerCoreLockedStacksArrayPool` backing field so the Jit can see the type